fe/fe-core/src/main/java/com/starrocks/planner/PlanNode.java

```diff
@@ -721,6 +721,10 @@ public boolean canUsePipeLine() {
         return false;
     }
 
+    public boolean canUseAdaptiveDop() {
```

Comment:
most of PlanNode's canUseAdaptiveDop is true, can we provide canUseAdaptiveDop method in PlanNode satisifies：
1. Leaf PlanNode always return true;
2. Non-leaf PlanNode depends its children.

for derived PlanNodes:
1. Only derived PlanNodes can not use AdpativeDop return false.
2. ExchangeNode return true.

Reply:
I worry that when a new derived `PlanNode` class is added and the adaptive DOP strategy doesn't support it, then there will be something wrong if it forget to make `canUseAdaptiveDop` return false.

Reply:
Update.